test/unittest/tests/block/test_block_addr.cpp

```diff
@@ -0,0 +1,182 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+/*
+ * [block_addr]: block_addr.c
+ * This file unit tests block manager functions relating to packing and unpacking address cookies.
+ */
+
+#include "wt_internal.h"
+#include <catch2/catch.hpp>
+#include <iostream>
+#include <vector>
+
+void
+unpack_addr_cookie_and_check(const uint8_t *packed, uint32_t block_allocsize, wt_off_t pack_offset,
+  uint32_t pack_size, uint32_t pack_checksum)
+{
+    uint64_t offset = 0, size = 0, checksum = 0;
+    REQUIRE(__wt_vunpack_uint(&packed, 0, &offset) == 0);
+    REQUIRE(__wt_vunpack_uint(&packed, 0, &size) == 0);
+    REQUIRE(__wt_vunpack_uint(&packed, 0, &checksum) == 0);
+
+    // Adjust the unpacked values as the block manager also does this to avoid storing large
+    // offsets.
+    wt_off_t unpacked_offset = (size == 0) ? static_cast<wt_off_t>(offset) :
+                                             static_cast<wt_off_t>(offset + 1) * block_allocsize;
+    uint32_t unpacked_size =
+      (size == 0) ? static_cast<uint32_t>(size) : static_cast<uint32_t>(size) * block_allocsize;
+    uint32_t unpacked_checksum = static_cast<uint32_t>(checksum);
+
+    if (pack_size != 0) {
+        CHECK(unpacked_offset == pack_offset);
+        CHECK(unpacked_size == pack_size);
+        CHECK(unpacked_checksum == pack_checksum);
+    } else {
+        CHECK(unpacked_offset == 0);
+        CHECK(unpacked_size == 0);
+        CHECK(unpacked_checksum == 0);
+    }
+}
+
+// Test the block manager's pack function.
+static void
+test_pack_addr_cookie(uint8_t *pp, WT_BLOCK *block, size_t *addr_size, wt_off_t pack_offset,
+  uint32_t pack_size, uint32_t pack_checksum)
+{
+    const uint8_t *begin = static_cast<const uint8_t *>(pp);
```

Comment:
I think you can drop the cast entirely.

Reply:
We're doing this because the block manager pack functions require a non constant pointer while the block manager unpack functions require a const pointer.

Reply:
So you can still past a non-const pointer to a function requiring a const pointer. All that that means is that within that function which takes the const pointer the pointer variable cannot be mutated.

In the same way you can assign a non-const to a const with no cast, there is also the `const_cast` construct but typically that is used for const->non-const casting. 

In all cases you can drop the static_cast, and in the function `test_pack_and_unpack_addr_cookie` `begin` is not required.

Reply:
Ah ok, thanks for clarifying. I'll give that a go.

Reply:
Fixed this in the most recent commit.